lib/vsc/filesystem/lustre.py

```diff
@@ -0,0 +1,613 @@
+        """
+
+        cmd = ['/usr/bin/lfs', name]
+
+        if opts is not None:
+            if isinstance(opts, (tuple, list,)):
+                cmd += list(opts)
+            else:
+                self.log.raiseException("_execute_lfs: please use a list or tuple for options: cmd %s opts %s" %
+                                        (cmd, opts), LustreOperationError)
+
+        ec, out = self._execute(cmd, changes)
+
+        return ec, out
+
+    def _execute_lctl_get_param_qmt_yaml(self, device, typ, quotyp='block'):
+        """ executy LUSTRE lctl get_param qmt.* command and parse output
+
+        eg:
+        lctl get_param qmt.kwlust-*.dt*.glb-prj
+        qmt.kwlust-QMT0000.dt-0x0.glb-prj=
+        global_pool0_dt_prj
+        - id:      0
+          limits:  { hard:     0, soft:   0, granted:    0, time:    604800 }
+        - id:      1
+          limits:  { hard:  3798016, soft:   3591168, granted:   3874836, time:  1599748949 }
+        - id:      2
+          limits:  { hard:   0, soft:   0, granted:  0, time:   281474976710656 }
+        - id:      3
+          limits:  { hard:   0, soft:   0, granted:  0, time: 281474976710656 }
+
+        """
+        if typ not in TYP2PARAM:
+            self.log.raiseException("_execute_lctl_get_param_qmt_yaml: unsupported type %s. Use USR,GRP or FILESET"
+                % typ, LustreOperationError)
+        if quotyp not in QUOTYP2PARAM:
+            self.log.raiseException("_execute_lctl_get_param_qmt_yaml: unsupported type %s. Use 'block' or 'inode'" %
+                quotyp, LustreOperationError)
+
+        param = 'qmt.%s-*.%s-*.glb-%s' % (device, QUOTYP2PARAM[quotyp], TYP2PARAM[typ])
+        opts = ['get_param', param]
+        _ec, res = self._execute_lctl(opts)
+        quota_info = res.split("\n", 2)
+        try:
+            newres = yaml.safe_load(quota_info[2])
+        except yaml.YAMLError as exc:
+            self.log.raiseException("_execute_lctl_get_param_qmt_yaml: Error in yaml output: %s"
+                % exc, LustreOperationError)
+
+        return newres
+
+    def _execute_lctl(self, opts, changes=False):
+        """ Return output of lctl command """
+
+        cmd = ['/usr/sbin/lctl']
+        if opts is not None:
+            if isinstance(opts, (tuple, list,)):
+                cmd += list(opts)
+            else:
+                self.log.raiseException("_execute_lctl: please use a list or tuple for options: cmd %s opts %s" %
+                                        (cmd, opts), LustreOperationError)
+
+        ec, res = self._execute(cmd, changes)
+
+        return ec, res
+
+    def list_filesystems(self, device=None):
+        """ List all Lustre file systems """
+        if not self.localfilesystems:
+            self._local_filesystems()
+
+        if device is None:
+            devices = []
+        elif not isinstance(device, list):
+            devices = [device]
+        else:
+            devices = device
```

Comment:
You're right, and I'll certainly do this for the private functions. This is copied from gpfs module and kept because I didn't check how it is called in vsc-administration/quota/whatever :)